scripts/west_commands/runners/nrfjprog.py

```diff
@@ -83,7 +83,9 @@ def get_board_snr(self, glob):
         # multiple boards and we are connected to a terminal, in which
         # case use print() and input() to ask what the user wants.
 
-        re_glob = escape(glob).replace(r"\*", ".+")
+        # Turn glob into a regex. Also, remove leading 0s for when the snr comes
```

Comment:
Are you 100% sure this is safe? None of our boards have semantically significant leading zeros in their IDs?

Reply:
Well, I would say 98%, but not 100% :P. Anyways, I changed it to be solved a different way now.